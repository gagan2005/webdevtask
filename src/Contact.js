// @flow
import { useMutation } from '@apollo/client';
import gql from 'graphql-tag';
import React, { useState } from 'react'
import Right from './Right';

// import catndog from './assets/Cat-N-Dog-Logo copy.png'
const submitquery = gql`
mutation sendMutation(
    $message: String!
    $email: String!
  ){
    sendMessage( email :$email , message :$message){
        email
    }
}`;

function Contact(props) {

    const [name, setName] = useState("");
    const [email, setEmail] = useState("");
    const [message, setMessage] = useState("");
    const [submitform] = useMutation(submitquery, {
        variables: {
            name: name,
            email: email,
            message: message
        }
        ,
        onError:(error)=>{
            setError(error.message);
             console.log(Object.keys(error));

            
        },
        onCompleted:(error)=>{
            setError("Succesful");
        }
    });
    const [error , setError] = useState("");

    
    if (props.disp == true)
        return (
            <div style={{ height: '100vh', width: '100vw', position: 'absolute' }} className="darken">
                <div className="contactus">
                    <Cancelicon toggle={props.toggle} />
                    <h3 style={{ color: '#495057', fontWeight: 'bold' }}>Contact</h3>
                    <span style={{ color: "#ADB5BD", fontSize: "12px" }}>
                        Lorem Ipsum is simply dummy text of the printing
                        </span>
                    <br></br><br></br>
                    <Inpele label="Name" text="Enter Your name" var={name} funn={setName} />


                    <Inpele label="Email" text="Enter Your email" var={email} funn={setEmail} />
                    <Inpele label="Message" text="Your message" var={message} funn={setMessage} />

                    <button className="submit" onClick={() => submitform()}>Submit</button>
                    <span className="error">{error}</span>
                </div>
            </div>
        )
    else return <></>

}

function Inpele(props) {
    return (
        <div className="inpele">
            <div>{props.label}</div>
            <input placeholder={props.text} onChange={
                (e) => {
                    props.funn(e.target.value)
                }
            }></input>
        </div>);
}

function Cancelicon(props) {
    return (
        <div className="cancelicon" style={{ float: "right" }} onClick={props.toggle}>
            <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" className="bi bi-x-circle-fill" viewBox="0 0 16 16">
                <path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM5.354 4.646a.5.5 0 1 0-.708.708L7.293 8l-2.647 2.646a.5.5 0 0 0 .708.708L8 8.707l2.646 2.647a.5.5 0 0 0 .708-.708L8.707 8l2.647-2.646a.5.5 0 0 0-.708-.708L8 7.293 5.354 4.646z" />
            </svg>
        </div>
    )
}


export default Contact;