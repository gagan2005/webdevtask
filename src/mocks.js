import {gql} from '@apollo/client';

const submitquery = gql`
mutation sendMutation(
    $message: String!
    $email: String!
  ){
    sendMessage( email :$email , message :$message){
        email
    }
}`;
const mocks = [
  {
    request: {
      query: submitquery,
      variables: {
       email : "",
       message : " ",
      },
    },
    result: {
      data: {
        email:""
      },
    },
  },
];

export {mocks , submitquery}