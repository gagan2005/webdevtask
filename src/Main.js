// @flow
import React from 'react'
import Right from './Right'
import Left from './Left'
import Nav from './Nav'
import Contact from './Contact';
export default class Main extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            disp_popup: false
        }
        this.toggle_popup = this.toggle_popup.bind(this);

    }

    toggle_popup() {
        this.setState((prevstate) => {
            return {
                disp_popup:!(prevstate.disp_popup)

            }

        })
    }
    render() {
        let darkening = "";
        if (this.state.disp_popup == true) darkening = "darken";
        return (<>
            <Contact disp={this.state.disp_popup} toggle={this.toggle_popup} />
            <div >
                <Nav action={this.toggle_popup} />
                <div className="container-fluid main-container">
                    <div className="row">
                        <Left />
                        <Right />
                    </div>
                </div>
            </div>
        </>

        );
    }
}