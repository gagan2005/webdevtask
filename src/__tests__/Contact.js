import React from 'react';
import { shallow ,mount } from 'enzyme';
import { MockedProvider } from '@apollo/client/testing';
import {gql} from '@apollo/client';

import Contact from '../Contact.js';
import Main from '../Main.js';
import {mocks , submitquery} from "../mocks.js"

describe('Contact Component', () => {

  it('Should be not visible when disp=false', 
  () =>
   {    
    
    const component = mount(
      <MockedProvider mocks={mocks}>
    < Contact disp={false} />
    </MockedProvider>);
  
    expect(component.find(Contact)).toMatchSnapshot();
  });

  it('Should be  visible when disp=True', 
  () =>
   {    
    
    const component = mount(<MockedProvider mocks={mocks}>
     < Contact disp={true} />
     </MockedProvider>);
  
    expect(component.find(Contact)).toMatchSnapshot();
  });
});

