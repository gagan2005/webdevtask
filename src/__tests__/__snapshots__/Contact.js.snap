// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component Should be  visible when disp=True 1`] = `
<Contact
  disp={true}
>
  <div
    className="darken"
    style={
      Object {
        "height": "100vh",
        "position": "absolute",
        "width": "100vw",
      }
    }
  >
    <div
      className="contactus"
    >
      <Cancelicon>
        <div
          className="cancelicon"
          style={
            Object {
              "float": "right",
            }
          }
        >
          <svg
            className="bi bi-x-circle-fill"
            fill="currentColor"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM5.354 4.646a.5.5 0 1 0-.708.708L7.293 8l-2.647 2.646a.5.5 0 0 0 .708.708L8 8.707l2.646 2.647a.5.5 0 0 0 .708-.708L8.707 8l2.647-2.646a.5.5 0 0 0-.708-.708L8 7.293 5.354 4.646z"
            />
          </svg>
        </div>
      </Cancelicon>
      <h3
        style={
          Object {
            "color": "#495057",
            "fontWeight": "bold",
          }
        }
      >
        Contact
      </h3>
      <span
        style={
          Object {
            "color": "#ADB5BD",
            "fontSize": "12px",
          }
        }
      >
        Lorem Ipsum is simply dummy text of the printing
      </span>
      <br />
      <br />
      <Inpele
        funn={[Function]}
        label="Name"
        text="Enter Your name"
        var=""
      >
        <div
          className="inpele"
        >
          <div>
            Name
          </div>
          <input
            onChange={[Function]}
            placeholder="Enter Your name"
          />
        </div>
      </Inpele>
      <Inpele
        funn={[Function]}
        label="Email"
        text="Enter Your email"
        var=""
      >
        <div
          className="inpele"
        >
          <div>
            Email
          </div>
          <input
            onChange={[Function]}
            placeholder="Enter Your email"
          />
        </div>
      </Inpele>
      <Inpele
        funn={[Function]}
        label="Message"
        text="Your message"
        var=""
      >
        <div
          className="inpele"
        >
          <div>
            Message
          </div>
          <input
            onChange={[Function]}
            placeholder="Your message"
          />
        </div>
      </Inpele>
      <button
        className="submit"
        onClick={[Function]}
      >
        Submit
      </button>
      <span
        className="error"
      />
    </div>
  </div>
</Contact>
`;

exports[`Contact Component Should be not visible when disp=false 1`] = `
<Contact
  disp={false}
/>
`;
